projects/packages/jetpack-mu-wpcom/src/features/wpcom-themes/wpcomsh-themes.php

```diff
@@ -0,0 +1,189 @@
+<?php
+/**
+ * Registers a filter for the themes API result to add WPCom themes.
+ *
+ * @package wpcom-themes
+ */
+
+/**
+ * Loads the WPCom themes service instance.
+ *
+ * @return WPCom_Themes_Service
+ */
+function wpcomsh_get_wpcom_themes_service_instance(): WPCom_Themes_Service {
+	static $instance;
+	if ( ! $instance ) {
+		// Load dependencies.
+		require_once __DIR__ . '/includes/class-wpcom-themes-mapper.php';
+		require_once __DIR__ . '/includes/class-wpcom-themes-merger.php';
+		require_once __DIR__ . '/includes/class-wpcom-themes-cache.php';
+		require_once __DIR__ . '/includes/class-wpcom-themes-service.php';
+		require_once __DIR__ . '/includes/class-wpcom-themes-api.php';
+
+		// Resolve and Inject dependencies.
+		$mapper               = new WPCom_Themes_Mapper();
+		$merger               = new WPCom_Themes_Merger();
+		$cache                = new WPCom_Themes_Cache();
+		$api                  = new WPCom_Themes_Api( $cache );
+		$wpcom_themes_service = new WPCom_Themes_Service( $api, $mapper, $merger );
+
+		$instance = $wpcom_themes_service;
+	}
+
+	return $instance;
+}
+
+/**
+ * Process the themes API result and add the recommended WPCom themes to the Popular tab.
+ *
+ * @param mixed  $res    The result object.
+ * @param string $action The action.
+ * @param mixed  $args   The arguments.
+ *
+ * @return mixed|stdClass
+ */
+function wpcomsh_popular_wpcom_themes_api_result( $res, string $action, $args ) {
+	// Pre-requisites checks.
+	$browse = $args->browse ?? '';
+	if ( 'query_themes' !== $action || 'popular' !== $browse ) {
+		return $res;
+	}
+
+	$wpcom_themes_service = wpcomsh_get_wpcom_themes_service_instance();
+
+	// Add results to the resulting array.
+	return $wpcom_themes_service->filter_themes_api_result_recommended( $res );
+}
+add_filter( 'themes_api_result', 'wpcomsh_popular_wpcom_themes_api_result', 0, 3 );
```

Comment:
We will want these to also run on WPcom in the [untangling themes PR](https://github.com/Automattic/wp-calypso/issues/98661), right? Should we move them to run on WPcom now, or only in the untangling themes PR?

Reply:
I a bit torn on if we should rename these functions and move the filters over entirely or not. I _think_ it's best to leave it as is. We don't need any of this on WPcom _yet_. It's probably best to enable it in untangling themes as that's when it's actually needed.

Reply:
Yeah I agree. If we keep this PR as just a code moving task, then we can consider changes in the future on a case by case basis. If we start changing things as well as moving them its gets harder to reason about.

Reply:
Actually, when I thought about the deploy process for this - because wpcomsh and jetpack-mu-wpcom are deployed at different times, I think it does make more sense to rename the functions now - that way we avoid errors when a function is redefined.

Reply:
I was worried about this too, but I think it's not an issue. I believe WoA sites run jetpack-mu-wpcom bundled as wpcomsh. They don't run the WPcom mu-plugin AND wpcomsh. So there should only be one version of anything running, even if deploy cycles are different. We should absolutely confirm to be sure though!

Reply:
Yeah I wasn't sure exactly how they fit together but @zinigor suggested it might be a problem and I thought better safe than sorry, since we want to rename anyway.

Reply:
>  I believe WoA sites run jetpack-mu-wpcom bundled as wpcomsh. They don't run the WPcom mu-plugin AND wpcomsh.

That's correct! I sometimes get confused with what environment gets affected with what. WoA and Atomic sites should be fine in this regard. The WordPress.com environment though will have these files removed, and I believe they were never used there in the first place, were they?

If that's so, then I'm sorry for causing additional anxiety about deploying this.